File: mythos/norse/audit_ascii_art.py
#!/usr/bin/env python3
"""
Audit script to find pages with ASCII art instead of SVG.
"""
import os
import re

def get_all_html_files(base_dir):
    """Get all HTML files in the Norse directory."""
    files = []
    for root, dirs, filenames in os.walk(base_dir):
        for filename in filenames:
            if filename.endswith('.html'):
                files.append(os.path.join(root, filename))
    return files

def check_for_ascii_art(html_file):
    """Check if file contains ASCII art."""
    try:
        with open(html_file, 'r', encoding='utf-8') as f:
            content = f.read()
    except Exception as e:
        return None, f"Error reading file: {e}"

    # Look for common ASCII art patterns
    ascii_indicators = [
        r'<pre[^>]*>[\s\S]*?[|/\\+\-]{3,}[\s\S]*?</pre>',  # Box drawings
        r'<pre[^>]*>[\s\S]*?\n[\s]*[|/\\][\s\S]*?\n[\s]*[|/\\][\s\S]*?</pre>',  # Multiple lines with art chars
        r'<pre[^>]*>[\s\S]*?╔═╗[\s\S]*?</pre>',  # Unicode box drawing
        r'<pre[^>]*>[\s\S]*?┌─┐[\s\S]*?</pre>',  # Unicode box drawing
    ]

    has_ascii = False
    ascii_snippets = []

    for pattern in ascii_indicators:
        matches = re.finditer(pattern, content, re.MULTILINE)
        for match in matches:
            snippet = match.group(0)[:200]  # First 200 chars
            has_ascii = True
            ascii_snippets.append(snippet)

    # Check for SVG
    has_svg = bool(re.search(r'<svg[^>]*>', content))

    return {
        'has_ascii_art': has_ascii,
        'has_svg': has_svg,
        'ascii_snippets': ascii_snippets
    }, None

def main():
    base_dir = os.path.dirname(os.path.abspath(__file__))
    html_files = get_all_html_files(base_dir)

    has_ascii = []
    needs_svg = []
    errors = []

    print(f"Checking {len(html_files)} HTML files for ASCII art...\n")

    for html_file in html_files:
        rel_path = os.path.relpath(html_file, base_dir)
        result, error = check_for_ascii_art(html_file)

        if error:
            errors.append({'file': rel_path, 'error': error})
            continue

        if result['has_ascii_art']:
            has_ascii.append({
                'file': rel_path,
                'has_svg': result['has_svg'],
                'snippets': result['ascii_snippets']
            })

            if not result['has_svg']:
                needs_svg.append(rel_path)

    if has_ascii:
        print(f"FILES WITH ASCII ART ({len(has_ascii)} files):")
        for item in has_ascii:
            print(f"\n  File: {item['file']}")
            print(f"  Has SVG: {item['has_svg']}")
            if item['snippets']:
                print(f"  Sample: {item['snippets'][0][:100]}...")
        print()

    if needs_svg:
        print(f"FILES NEEDING SVG CONVERSION ({len(needs_svg)} files):")
        for f in needs_svg:
            print(f"  - {f}")
        print()

    if errors:
        print(f"ERRORS ({len(errors)} files):")
        for item in errors:
            print(f"  - {item['file']}: {item['error']}")
        print()

    if not has_ascii:
        print("NO ASCII ART FOUND!")

    return len(needs_svg)

if __name__ == '__main__':
    exit(main())

File: mythos/chinese/audit-ascii-art.js
#!/usr/bin/env node

/**
 * Audit script: Find ASCII art that should be replaced with SVG/Unicode
 * Detects old-style ASCII diagrams in Chinese mythology pages
 */

const fs = require('fs');
const path = require('path');

const chineseDir = path.join(__dirname);
const filesWithAsciiArt = [];

// Recursively get all HTML files
function getHtmlFiles(dir) {
    const files = [];
    const items = fs.readdirSync(dir);

    for (const item of items) {
        const fullPath = path.join(dir, item);
        const stat = fs.statSync(fullPath);

        if (stat.isDirectory()) {
            files.push(...getHtmlFiles(fullPath));
        } else if (item.endsWith('.html')) {
            files.push(fullPath);
        }
    }

    return files;
}

// Detect ASCII art patterns
function detectAsciiArt(content) {
    const asciiPatterns = [
        // Box drawing characters
        /[─│┌┐└┘├┤┬┴┼═║╔╗╚╝╠╣╦╩╬]+/g,
        // Multiple consecutive dashes/underscores/pipes
        /[-_|]{5,}/g,
        // ASCII art trees/diagrams
        /\s+[\/\\|]+\s+/g,
        // Pre-formatted ASCII blocks
        /<pre>[\s\S]*?[─│┌┐└┘├┤┬┴┼|\/\\]{3,}[\s\S]*?<\/pre>/g
    ];

    const findings = [];

    for (const pattern of asciiPatterns) {
        const matches = content.match(pattern);
        if (matches && matches.length > 0) {
            findings.push(...matches);
        }
    }

    return findings;
}

// Check for modern SVG/emoji alternatives
function hasModernVisuals(content) {
    return {
        hasSvg: content.includes('<svg'),
        hasUnicodeSymbols: /[☯️🌙☀️🌊🏔️⚔️💧🔥🌳]/.test(content),
        hasHeroIcons: content.includes('hero-icon-display'),
        hasDeityIcons: content.includes('deity-icon')
    };
}

// Main audit
function auditAsciiArt() {
    console.log('🎨 Auditing for ASCII art in Chinese mythology section...\n');

    const htmlFiles = getHtmlFiles(chineseDir);
    console.log(`Found ${htmlFiles.length} HTML files to audit\n`);

    for (const file of htmlFiles) {
        const content = fs.readFileSync(file, 'utf-8');
        const relativePath = path.relative(chineseDir, file);
        const asciiArt = detectAsciiArt(content);
        const modernVisuals = hasModernVisuals(content);

        if (asciiArt.length > 0) {
            filesWithAsciiArt.push({
                file: relativePath,
                asciiCount: asciiArt.length,
                modernVisuals,
                samples: asciiArt.slice(0, 3) // First 3 examples
            });
        }
    }

    // Report results
    console.log('━'.repeat(80));
    console.log('AUDIT RESULTS: ASCII Art Detection');
    console.log('━'.repeat(80));
    console.log(`Total files checked: ${htmlFiles.length}`);
    console.log(`Files with ASCII art: ${filesWithAsciiArt.length}\n`);

    if (filesWithAsciiArt.length > 0) {
        console.log('⚠️  FILES WITH ASCII ART:\n');

        for (const result of filesWithAsciiArt) {
            console.log(`\n📄 ${result.file}`);
            console.log(`   ASCII patterns found: ${result.asciiCount}`);
            console.log(`   Samples: ${result.samples.join(', ').substring(0, 80)}...`);
            console.log(`   Modern alternatives:`);
            console.log(`     - SVG graphics: ${result.modernVisuals.hasSvg ? '✅' : '❌'}`);
            console.log(`     - Unicode symbols: ${result.modernVisuals.hasUnicodeSymbols ? '✅' : '❌'}`);
            console.log(`     - Hero icons: ${result.modernVisuals.hasHeroIcons ? '✅' : '❌'}`);
            console.log(`     - Deity icons: ${result.modernVisuals.hasDeityIcons ? '✅' : '✅ (using emoji)'}`);
        }

        console.log('\n' + '━'.repeat(80));
        console.log(`\n💡 Recommendation: Replace ASCII art with Unicode emoji (☯️, 🐉, etc.) or SVG\n`);
    } else {
        console.log('✅ No ASCII art detected! All visuals use modern formats.\n');
    }
}

// Run audit
try {
    auditAsciiArt();
} catch (error) {
    console.error('Error running audit:', error);
    process.exit(1);
}
